src/data/polynomial/eval.lean

```diff
@@ -507,6 +507,13 @@ begin
   split_ifs; simp [f.map_zero],
 end
 
+@[simps] def map_equiv (e : R ≃+* S) : polynomial R ≃+* polynomial S :=
+ring_equiv.of_hom_inv
+  (map_ring_hom e)
+  (map_ring_hom e.symm)
+  (by ext; simp)
+  (by ext; simp)
```

Comment:
I was hoping I could golf this proof with these lemmas:
```
lemma map_map [semiring T] (g : S →+* T)
  (p : polynomial R) : (p.map f).map g = p.map (g.comp f) :=
ext (by simp [coeff_map])

@[simp] lemma map_id (p : polynomial R) : p.map (ring_hom.id _) = p :=
by simp [polynomial.ext_iff, coeff_map]

lemma map_ring_hom_comp [semiring T] (g : S →+* T) :
  (map_ring_hom g).comp (map_ring_hom f) = map_ring_hom (g.comp f) :=
ring_hom.ext (map_map _ _)

@[simp] lemma map_ring_hom_id : map_ring_hom (ring_hom.id R) = ring_hom.id (polynomial R) :=
ring_hom.ext map_id
```
but the coercions made it awkward. Do you want to add these anyway, or would you prefer to merge as is?